sound/soc/sof/sof-audio.c

```diff
@@ -127,6 +172,12 @@ static int sof_widget_setup(struct snd_sof_dev *sdev, struct snd_sof_widget *swi
 		dev_err(sdev->dev, "error: failed to load widget type %d with ID: %d\n",
 			swidget->widget->id, swidget->comp_id);
 
+	/* restore kcontrols for widget */
+	ret = sof_widget_kcontrol_setup(sdev, swidget);
```

Comment:
firstly you're overwriting potential error from above, secondly is this needed `if (ret < 0)` from above?